activerecord/test/cases/dirty_test.rb

```diff
@@ -941,6 +941,51 @@ def check_around
     assert_not_predicate person, :changed?
   end
 
+  test "partial insert off with unchanged default function attribute" do
+    partial_inserts_was = Aircraft.partial_inserts
+    Aircraft.partial_inserts = false
+    Aircraft.connection.add_column(:aircraft, :manufactured_at, :timestamp, default: -> { "now()" })
+
+    aircraft = nil
+    assert_sql(/(?!manufactured_at)/i) do
```

Comment:
I'm not sure this is a good assertion. You can hope that there's only a single query emitted, in which case it does work, but if for some reason the driver emit multiple queries, then it doesn't test anywhing because `assert_sql` is basically a `any?` check.

It's also a bit redundant with `ActiveRecord::SQLCounter.log_all.none?`

Reply:
@byroot @intrip Can you suggest the correct assertion to be used for this? I tried `ActiveRecord::SQLCounter.log_all.none?` but I am not able to correctly assert the absence of the `manufactured_at` column in the insert query.

Reply:
I'd give up trying to assert the emitted query directly, and just test the actual behavior. e..g assert the unsaved model has no `manufactured_at`, save, assert it has one.